services/management/ia/channels_service.go

```diff
@@ -17,15 +17,270 @@
+					Channel:      config.Channel,
+				},
+			}
+		case models.WebHook:
+			config := channel.WebHookConfig
+			c.Channel = &iav1beta1.Channel_WebhookConfig{
+				WebhookConfig: &iav1beta1.WebhookConfig{
+					SendResolved: config.SendResolved,
+					Url:          config.URL,
+					HttpConfig:   convertModelToHTTPConfig(config.HTTPConfig),
+					MaxAlerts:    config.MaxAlerts,
+				},
+			}
+		default:
+			return nil, errors.Wrapf(err, "unknown notification channel type %s", channel.Type)
+		}
+
+		res[i] = c
+	}
+
+	return &iav1beta1.ListChannelsResponse{Channels: res}, nil
+}
+
+// AddChannel adds new notification channel.
+func (s *ChannelsService) AddChannel(ctx context.Context, req *iav1beta1.AddChannelRequest) (*iav1beta1.AddChannelResponse, error) {
+	channel := &models.Channel{
+		ID:       req.GetChannelId(),
+		Disabled: req.GetDisabled(),
+	}
+
+	if emailConf := req.GetEmailConfig(); emailConf != nil {
+		channel.Type = models.Email
+		channel.EmailConfig = &models.EmailConfig{
+			SendResolved: emailConf.SendResolved,
+			To:           emailConf.To,
+		}
+	}
+	if pagerDutyConf := req.GetPagerdutyConfig(); pagerDutyConf != nil {
+		if channel.Type != "" {
+			return nil, status.Error(codes.InvalidArgument, "Request should contain only one type of channel configuration")
+		}
+		channel.Type = models.PagerDuty
+		channel.PagerDutyConfig = &models.PagerDutyConfig{
+			SendResolved: pagerDutyConf.SendResolved,
+			RoutingKey:   pagerDutyConf.RoutingKey,
+			ServiceKey:   pagerDutyConf.ServiceKey,
+		}
+	}
+	if slackConf := req.GetSlackConfig(); slackConf != nil {
+		if channel.Type != "" {
+			return nil, status.Error(codes.InvalidArgument, "Request should contain only one type of channel configuration")
+		}
+		channel.Type = models.Slack
+		channel.SlackConfig = &models.SlackConfig{
+			SendResolved: slackConf.SendResolved,
+			Channel:      slackConf.Channel,
+		}
+	}
+	if webhookConf := req.GetWebhookConfig(); webhookConf != nil {
+		if channel.Type != "" {
+			return nil, status.Error(codes.InvalidArgument, "Request should contain only one type of channel configuration")
+		}
+		channel.Type = models.WebHook
+		channel.WebHookConfig = &models.WebHookConfig{
+			SendResolved: webhookConf.SendResolved,
+			URL:          webhookConf.Url,
+			MaxAlerts:    webhookConf.MaxAlerts,
+			HTTPConfig:   convertHTTPConfigToModel(webhookConf.HttpConfig),
+		}
+	}
+
+	err := s.ia.AddChannel(channel)
+	if err != nil {
+		return nil, err
+	}
+
+	return &iav1beta1.AddChannelResponse{}, nil
 }
 
-func NewChannelsService() *ChannelsService {
-	return &ChannelsService{}
+// ChangeChannel changes existing notification channel.
+func (s *ChannelsService) ChangeChannel(ctx context.Context, req *iav1beta1.ChangeChannelRequest) (*iav1beta1.ChangeChannelResponse, error) {
+	channel := &models.Channel{
+		ID:       req.GetChannelId(),
+		Disabled: req.GetDisabled(),
+	}
+
+	if emailConf := req.GetEmailConfig(); emailConf != nil {
```

Comment:
maybe it's better to use switch here and other places?

Reply:
It's different fields/methods, there is nothing to switch over.